framework/bucket_policies/access_validation_strategy_factory.py

```diff
@@ -0,0 +1,59 @@
+import importlib
+
+from utility.utils import camel_to_snake
+
+
+class AccessValidationStrategyFactory:
+    """
+    A factory for creating AccessValidationStrategy instances
+    """
+
+    @staticmethod
+    def create_strategy_for_operation(admin_client, bucket, operation):
+        """
+        Create an AccessValidationStrategy instance for the given operation.
+        Dynamically imports the appropriate strategy class and module based on the operation,
+        and returns an instance of that class.
+
+        Supported operations need to have a corresponding strategy class
+        defined the access_validation_strategies module so that the factory can create an instance of it.
+        The file name should start with the operation name in snake case, followed by "_validation_strategy.py".
+        The class name should be the operation name in camel case, followed by "ValidationStrategy".
+
+        i.e. for the operation "CopyObject", the file should be "copy_object_validation_strategy.py",
+        and the class should be "CopyObjectValidationStrategy".
+
+        Args:
+            admin_client (S3Client): The privileged client for setting up preconditions
+            bucket (str): The bucket to test access on
+            operation (str): The operation to test
+
+        Returns:
+            AccessValidationStrategy: An instance of the appropriate strategy
+
+        Raises:
+            NotImplementedError: If the operation is not supported
+        """
+
+        # Dynamically import the strategy module
+        try:
+            prefix = "framework.bucket_policies.access_validation_strategies."
```

Comment:
as discussed in person, please change to a dynamic way of setting the prefix.
https://github.com/red-hat-storage/ocs-ci/blob/master/ocs_ci/deployment/factory.py should be useful for reference

Not a PR merge blocker, so you can do that in a follow up PR

Reply:
Addressed in https://github.com/red-hat-storage/noobaa-sa-ci/pull/33, specifically in commit https://github.com/red-hat-storage/noobaa-sa-ci/pull/33/commits/dacf4a0fd50f8dbf7d34dc26e525778a32cb2a3b